src/Riak.php

```diff
@@ -213,10 +226,15 @@ public function getLastRequest()
      * @return $this
      * @throws Exception
      */
-    protected function pickNewNode()
+    public function pickNewNode()
     {
-        // mark current active node as inactive
+        // mark current active node as inactive and increment attempts
         $this->getActiveNode()->setInactive(true);
+        $this->attempts++;
+        $this->inactiveNodes[$this->getActiveNodeIndex()] = $this->getActiveNode();
```

Comment:
OK thanks for the info. I read [this doc](http://php.net/manual/en/features.persistent-connections.php) to get a bit more information about how PHP works.

Is there any way to persist node state across requests, so you can remember the inactive nodes and not re-try on the next HTTP request? I envision a scenario where quite a bit of time is lost sending requests to a persistently-down node. Of course, this requires a method to detect when a node has recovered, too. Just something to think about.